src/qililab/instruments/quantum_machines/quantum_machines_cluster.py

```diff
@@ -282,34 +285,85 @@ def reset(self):
         """Turns off an instrument."""
-        if self._is_connected_to_qm():
+        if self._is_connected_to_qm:
             self._qm.close()
+            self._is_connected_to_qm = False
 
-    def update_configuration(self, compilation_config: dict):
+    def append_configuration(self, configuration: dict):
         """Update the configuration dictionary by appending the configuration generated by compilation."""
-        self._config = merge_dictionaries(dict(self._config), compilation_config)
-        # If we are already connected, reopen the connection with the new configuration
-        if self._is_connected_to_qm():
-            self._qm = self._qmm.open_qm(config=self._config, close_other_machines=True)
+        merged_configuration = merge_dictionaries(dict(self._config), configuration)
+        if self._config != merged_configuration:
+            self._config = cast(DictQuaConfig, merged_configuration)
+            # If we are already connected, reopen the connection with the new configuration
+            if self._is_connected_to_qm:
+                self._qm = self._qmm.open_qm(config=self._config, close_other_machines=True)
 
     def run_octave_calibration(self):
         """Run calibration procedure for the buses with octaves, if any."""
         elements = [element for element in self._config["elements"] if "RF_inputs" in self._config["elements"][element]]
         for element in elements:
             self._qm.calibrate_element(element)
 
-    def set_parameter(self, parameter: Parameter, value: float | str | bool, channel_id: int | None = None):
-        """Sets the parameter of a specific instrument.
+    def set_parameter_of_bus(self, bus: str, parameter: Parameter, value: float | str | bool):
+        """Sets the parameter of the instrument
 
         Args:
-            parameter (Parameter): parameter settings of the instrument to update
-            value (float | str | bool): value to update
-            channel_id (int | None, optional): instrument channel to update, if multiple. Defaults to None.
+            bus (str): The assossiated bus to change parameter
+            parameter (Parameter): The parameter to update
+            value (float | str | bool): The new value of the parameter
 
-        Returns:
-            bool: True if the parameter is set correctly, False otherwise
+        Raises:
+            NotImplementedError: Raised if not connected to Quantum Machines
+            ParameterNotFound: Raised if parameter does not exist
+        """
+        # TODO: This is a temporary solution. Implement again when QM API gets updated.
+        if not self._is_connected_to_qm:
+            raise NotImplementedError("You should be connected to Quantum Machines in order to change a parameter.")
+        if parameter == Parameter.LO_FREQUENCY:
+            self._qm.octave.set_lo_frequency(element=bus, lo_frequency=float(value))
+            if "RF_inputs" in self._config["elements"][bus]:
```

Comment:
if `parameter == Parameter.LO_FREQUENCY` but `("RF_inputs" in self._config["elements"][bus]) == False` then we will raise a `ParameterNotFound` error (but technically the parameter does exist). Should this happen?

Reply:
Same for other parameters below

Reply:
No, if `"RF_inputs" in self._config["elements"][bus]) == False` then you will just not update the `_config` dictionary. But you will not raise a `ParameterNotFound` error either. 

In general the implementation of `get_parameter` and `set_parameter` is super temporary until the QM API is improved (I had a meeting with Theo on that and they will try to make the changes soon)

This is why I don't want to mess with too many checks and exception raising. I assume that the ones who use it right now, know what they are doing. And also I assume we work only with Octave and not some external up/downconverter.

Reply:
Eventually, I re-implemented the method to be more robust, and also save the changes to `settings` as well.